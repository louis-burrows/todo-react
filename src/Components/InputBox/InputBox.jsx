import React, { useState, useEffect } from "react";
import styles from "./InputBox.module.scss";

const InputBox = (props) => {

  const { maxIdCount } = props;

  const [currentInput, captureInput] = useState("");
  const [idForItem, updateID] = useState(1);

  useEffect(() => {
    return maxIdCount ? updateID(maxIdCount+1) : "";
  }, [])

  const createItem = () => {
    props.updateList([
      ...props.totalListItems, 
      { toDo: currentInput, 
        isCompleted: false, 
        id: idForItem }
    ]);
    updateID(idForItem + 1);
  }

  return (
    <>
        <label className={styles.labelForInput} htmlFor="addToList">
          Type a list entry: 
        </label>
        
        <input 
          className={styles.inputBar}
          type="text" 
          name="addToList" 
          id="addToList" 
          onInput={e => captureInput(e.target.value)} />

        <button className={styles.submitButton} onClick={() => createItem()}> 
          Add Entry
        </button>
    </>
  );
};

export default InputBox;
