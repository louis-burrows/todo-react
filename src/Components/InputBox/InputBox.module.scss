input {
  margin-left: 10px;
  padding: 3px;
  border: 1px solid black;
  border-radius: 4px;
}

input[type="text"] {
  width: 300px;
}

input[type="submit"] {
  font-weight: 600;
}