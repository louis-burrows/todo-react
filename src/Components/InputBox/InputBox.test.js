import React from "react";
import { shallow } from "enzyme";
import InputBox from "./InputBox";

describe("InputBox tests", () => {

  let testComponent;

  beforeEach(() => {
    testComponent = shallow(<InputBox />);
  })

  it("should render", () => {
    expect(testComponent).toBeTruthy();
  });

});
