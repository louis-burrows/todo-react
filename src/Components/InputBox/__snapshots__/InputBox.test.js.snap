// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox tests should match the snapshot 1`] = `ShallowWrapper {}`;
